### PR TITLE
New Sample: Create mobile geodatabase

### PR DESCRIPTION
**Pull request by shubham7109**
## Description
PR to add a new Kotlin sample _"Create mobile geodatabase"_ in the `Edit & Manage Data` category.

## Links and Data
Sample Epic: `runtime/common-samples/issues/2993`

## What To Review
-  Review the code to make sure it is easy to follow like other samples on Android
- `README.md` and `README.metadata.json` files

## How to Test
Run the sample on the sample viewer or the repo.

### CODE REVIEW

File: kotlin/create-and-edit-mobile-geodatabases/src/main/java/com/esri/arcgisruntime/sample/createandeditmobilegeodatabases/MainActivity.kt [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+        featureAttributes["name"] = "User's location"
+        featureAttributes["sequence"] = sequence
+        featureAttributes["collection_timestamp"] = Date().time
+
+        // create a new feature
+        val feature = featureTable?.createFeature(featureAttributes, mapPoint)
+        // add the feature to the feature table
+        featureTable?.addFeatureAsync(feature)?.addDoneListener {
+            // feature added to the table
+            Log.e(TAG, "Added feature to table, size = ${featureTable!!.totalFeatureCount}")
```
Comment (shubham7109) [inline kotlin/create-and-edit-mobile-geodatabases/src/main/java/com/esri/arcgisruntime/sample/createandeditmobilegeodatabases/MainActivity.kt:172]: Here we have a sample app that creates a geodatabase and adds features to the table. 
Issue:  
I'm not able to see the feature being added to the table here. The total feature count returns 0 always.

File: kotlin/create-mobile-geodatabase/src/main/java/com/esri/arcgisruntime/sample/createmobilegeodatabase/MainActivity.kt [hunk 2]
```diff
@@ -0,0 +1,296 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.esri.arcgisruntime.sample.createmobilegeodatabase
+
+import android.annotation.SuppressLint
```
Comment (TADraeseke) [inline kotlin/create-mobile-geodatabase/src/main/java/com/esri/arcgisruntime/sample/createmobilegeodatabase/MainActivity.kt:19]:
```suggestion
```

File: kotlin/create-mobile-geodatabase/src/main/java/com/esri/arcgisruntime/sample/createmobilegeodatabase/MainActivity.kt [hunk 3]
```diff
@@ -0,0 +1,296 @@
...
+        activityMainBinding.featureCount
+    }
+
+    // feature table created using mobile geodatabase and added to the MapView
+    private var featureTable: GeodatabaseFeatureTable? = null
+
+    // mobile geodatabase used to create and store the feature attributes (LocationHistory.geodatabase)
+    private var geodatabase: Geodatabase? = null
+
+    @SuppressLint("ClickableViewAccessibility")
```
Comment (TADraeseke) [inline kotlin/create-mobile-geodatabase/src/main/java/com/esri/arcgisruntime/sample/createmobilegeodatabase/MainActivity.kt:80]: remove

File: kotlin/create-mobile-geodatabase/src/main/java/com/esri/arcgisruntime/sample/createmobilegeodatabase/MainActivity.kt [hunk 4]
```diff
@@ -0,0 +1,296 @@
...
+            val tableFuture = geodatabase?.createTableAsync(tableDescription)
+            if (tableFuture != null) {
+                setupMapFromGeodatabase(tableFuture)
+            } else{
+                showError("Error adding FieldDescriptions to the mobile geodatabase")
+            }
+        }
+    }
+
+    @SuppressLint("SetTextI18n")
```
Comment (TADraeseke) [inline kotlin/create-mobile-geodatabase/src/main/java/com/esri/arcgisruntime/sample/createmobilegeodatabase/MainActivity.kt:191]: remove

**Review by TADraeseke — CHANGES_REQUESTED** [review]
Nice work @shubham7109!

At review tomorrow can you talk us through the end point of ending at a sharing Intent? Seems a cool idea, I wonder if writing to the app's scoped storage is to restrictive?

Do we want to include the ability to load a new layer from newly made geodatabase or is that too contrived?

Other than that just highlighted some small stuff below.

Also, we'll need a landscape screenshot as normal.

Nice job!

File: kotlin/create-mobile-geodatabase/src/main/java/com/esri/arcgisruntime/sample/createmobilegeodatabase/MainActivity.kt [hunk 5]
```diff
@@ -0,0 +1,296 @@
...
+            // create the geodatabase file to add features into the feature table
+                createGeodatabase()
+            else
+                showError("Error loading MapView: ${mapView.map.loadError.message}")
+        }
+
+        viewTableButton.setOnClickListener {
+            // displays table dialog with the values in the feature table
+            displayTable()
+        }
```
[the remark this answers is not in the document]
Comment (shubham7109) [reply kotlin/create-mobile-geodatabase/src/main/java/com/esri/arcgisruntime/sample/createmobilegeodatabase/MainActivity.kt:114]: Do you mean something like this?
- displayTable()
    - inflate and attatch layout with ViewBinding
    -  add `viewTableButton.setOnClickListener { show() }`

I would be against doing this as we have to inflate the layout multiple times on each button press and add newer table layout rows as features are added to the map.

**Comment by shubham7109** [comment]
At review tomorrow can you talk us through the end point of ending at a sharing Intent? Seems a cool idea, I wonder if writing to the app's scoped storage is to restrictive?

- Let's leave it for the user to decide on how to handle the file by adding permissions to write to external storage. The sharing intent is helpful to share the file out of the device to another using Bluetooth/Email/Drive etc, which seems more aligned with the workflow of mobile geodatabases. 

Do we want to include the ability to load a new layer from newly made geodatabase or is that too contrived?
- This could be a feature in a future sample "Edit Mobile Geodatabases" sample, where layers could be added/removed from a geodatabase file.

**Review by saratchandrakarumuri — APPROVED** [review]
Sample is working as expected

File: kotlin/create-mobile-geodatabase/src/main/java/com/esri/arcgisruntime/sample/createmobilegeodatabase/MainActivity.kt (hunk 5 above)
[the remark this answers is not in the document]
Comment (TADraeseke) [reply kotlin/create-mobile-geodatabase/src/main/java/com/esri/arcgisruntime/sample/createmobilegeodatabase/MainActivity.kt:114]: Nevermind looks like the method's been deprecated and the practice is out of favor. Used to be able to add `android:onClick="methodName"` to xml buttons to call a specific method.

No worries, you're code is totally valid (and better practice!)

File: kotlin/create-mobile-geodatabase/src/main/java/com/esri/arcgisruntime/sample/createmobilegeodatabase/MainActivity.kt [hunk 6]
```diff
@@ -0,0 +1,302 @@
...
+        super.onCreate(savedInstanceState)
+        setContentView(activityMainBinding.root)
+
+        // authentication with an API key or named user is
+        // required to access basemaps and other location services
+        ArcGISRuntimeEnvironment.setApiKey(BuildConfig.API_KEY)
+
+        mapView.apply {
+            //  create a map with a topographic basemap and set the map to the mapview
+            this.map = ArcGISMap(BasemapStyle.ARCGIS_TOPOGRAPHIC)
```
Comment (TADraeseke) [inline kotlin/create-mobile-geodatabase/src/main/java/com/esri/arcgisruntime/sample/createmobilegeodatabase/MainActivity.kt:89]:
```suggestion
            map = ArcGISMap(BasemapStyle.ARCGIS_TOPOGRAPHIC)
```

File: kotlin/create-mobile-geodatabase/src/main/java/com/esri/arcgisruntime/sample/createmobilegeodatabase/MainActivity.kt [hunk 7]
```diff
@@ -0,0 +1,302 @@
...
+                mapView.map.operationalLayers.clear()
+                // disable the button since no features are displayed
+                viewTableButton.isEnabled = false
+                // create a new geodatabase file to add features into the feature table
+                createGeodatabase()
+            } else {
+                showError("Error loading MapView: ${mapView.map.loadError.message}")
+            }
+        }
+
```
Comment (TADraeseke) [inline kotlin/create-mobile-geodatabase/src/main/java/com/esri/arcgisruntime/sample/createmobilegeodatabase/MainActivity.kt:295]: remove whitespace

Comment (TADraeseke) [inline kotlin/create-mobile-geodatabase/src/main/java/com/esri/arcgisruntime/sample/createmobilegeodatabase/MainActivity.kt:295]: maybe better to extracted to it's own function called "resetMap" or something

File: kotlin/create-mobile-geodatabase/src/main/res/xml/provider.xml [hunk 8]
```diff
@@ -0,0 +1,4 @@
+<?xml version="1.0" encoding="utf-8"?>
+<paths>
+    <external-path name="share" path="." />
+</paths>
```
Comment (TADraeseke) [inline kotlin/create-mobile-geodatabase/src/main/res/xml/provider.xml:4]: missing return at end of file

**Review by TADraeseke — APPROVED** [review]
Great work @shubham7109. Good use case and very clean code 🎉 

I've left a few small comments below to be addressed before you merge please. I'll approve now though, so I don't hold things up.

Thanks!

**Pull request merged**